### ISSUE TITLE
p-datepicker screenreader accessibility issues

### ISSUE THREAD

**Issue by simo-leon-pelzer** [primefaces/primeng#19626]
### Describe the bug

Found several screenreader accessibility issues in the p-datepicker:

- The selection state of the day is not announced
- The current month is not announced when the month changes
- The days only announce their respective number (rather than the day, month and year)
- The accessible name for the month/year (“May”/"2026") is overridden with “Select month/select year"
- The right/left arrow in the right/left-hand column changes the month (instead of jumping to the next/previous day) (https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/)

### Reason for not contributing a PR

- [ ] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://primeng.org/datepicker

### Environment

Tested with VoiceOver & Edge on MacOs and Linux Mint, Firefox & Brave and Ocular. And verified with dev tools accessibility tree.

### Angular version

21.2.11

### PrimeNG version

v21

### Steps to reproduce the behavior

1. Open the datepicker with the keyboard
2. Start a screenreader

- Navigate through days with arrow keys
- Change a month or year with the keyboard

### Expected behavior

- The selection state of the day is not announced
- **Expected:** the selected day should be announced
- The current month is not announced when the month changes
- **Expected:** when the month changes via month select button, arrow buttons or arrow keys the changed month should be announced
- The days only announce their respective number (rather than the day, month and year)
- **Expected:** day, month and year should be announced
- The accessible name for the month/year (“May”/"2026") is overridden with “Select month/select year"
- **Expected:** the current selected month sould be announced too. Example: "Choose month, current May",  "Choose year, current 2026"
- The right/left arrow in the right/left-hand column changes the month (instead of jumping to the next/previous day)
- **Expected:** The right/left arrow in the right/left-hand column changes the day
